paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -417,36 +420,91 @@ internal class PaymentSheetViewModel @Inject internal constructor(
     }
 
     private fun confirmPaymentSelection(paymentSelection: PaymentSelection?) {
-        val clientSecret = when (val mode = args.initializationMode) {
-            is PaymentSheet.InitializationMode.PaymentIntent -> {
-                PaymentIntentClientSecret(mode.clientSecret)
-            }
-            is PaymentSheet.InitializationMode.SetupIntent -> {
-                SetupIntentClientSecret(mode.clientSecret)
-            }
-            is PaymentSheet.InitializationMode.DeferredIntent -> {
-                TODO("Not implemented yet")
+        viewModelScope.launch {
```

Comment:
Do you think it makes sense to pull this out into a separate class that fetches the payment intent (or calls the merchant code for the deferred flow) and then returns the intent’s status? Based on that information, we’ll be able to determine whether we need to confirm the intent or handle its next action.